### DISCUSSION TITLE
i/o timeout

### DISCUSSION THREAD

**Discussion by Bobbobmebob** [teslamate-org/teslamate#1022] [Q&A]
**Describe the bug**
Doing a fresh install and when running docker-compose up and it tries to pull the latest image and delivers a timeout

PS C:\teslamate> docker-compose up
Pulling teslamate (teslamate/teslamate:latest)...
ERROR: Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on 192.168.65.1:53: read udp 192.168.65.3:40463->192.168.65.1:53: i/o timeout
PS C:\teslamate>

**Expected behavior**
This worked yesterday, but this morning delivers the above error

**How to reproduce it (as minimally and precisely as possible)**:

**Relevant entries from the logs**

```
i/o timeout
```

**Screenshots**
error text copied above

**Data**

**Environment**

- TeslaMate version: trying to get latest for fresh install
- Type of installation: Docker or Manual: Docker
- OS TeslaMate is installed on: Win 10
- User OS & Browser: Win 10, Firefox/Chrome
- Others:

**Comment by ngardiner** [comment]
You have a network issue somewhere in your environment. 192.168.65.3 (your Docker machine) tries to do a DNS lookup of registry-1.docker.io on 192.168.65.1 (your apparent DNS resolver) and the request times out.

**Comment by Bobbobmebob** [comment]
My ip address range is 10.0.x.x so was assuming this is on the far end when it is trying to pull the container, but will double check.

**Comment by ngardiner** [comment]
It might be the internal Docker network range. Usually it's a 172.17 address but it can be changed. Not entirely sure how it works on Windows but on Linux it would be the IP subnet assigned to interface docker0.

**Comment by Bobbobmebob** [comment]
For people using Win 10, it appears to be partially related to this issue https://forums.docker.com/t/error-starting-userland-proxy-listen-tcp-0-0-0-0-bind-an-attempt-was-made-to-access-a-socket-in-a-way-forbidden-by-its-access-permissions/81299/5  
Stop IIS when running 'docker-compose up' Then restart after completion.

This allowed postgres to be pulled but stopped with timeout on pulling teslamate. The PC can resolve registry-1.docker.io to 52.5.11.128
PS C:\teslamate> docker-compose up
Creating network "teslamate_default" with the default driver
Creating volume "teslamate_teslamate-db" with default driver
Creating volume "teslamate_teslamate-grafana-data" with default driver
Creating volume "teslamate_mosquitto-conf" with default driver
Creating volume "teslamate_mosquitto-data" with default driver
Pulling database (postgres:13)...
13: Pulling from library/postgres
bb79b6b2107f: Pull complete
e3dc51fa2b56: Pull complete
f213b6f96d81: Pull complete
2780ac832fde: Pull complete
ae5cee1a3f12: Pull complete
95db3c06319e: Pull complete
475ca72764d5: Pull complete
8d602872ecae: Pull complete
c4fca31f2e3d: Pull complete
a00c442835e0: Pull complete
2e2305af3390: Pull complete
6cff852bb872: Pull complete
25bb0be11543: Pull complete
4738c099c4ad: Pull complete
Digest: sha256:8f7c3c9b61d82a4a021da5d9618faf056633e089302a726d619fa467c73609e4
Status: Downloaded newer image for postgres:13
Pulling teslamate (teslamate/teslamate:latest)...
ERROR: Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on 192.168.65.1:53: read udp 192.168.65.3:46752->192.168.65.1:53: i/o timeout

Have to investigate further. Have also tried to turning off firewall temporarily but that did not assist.
